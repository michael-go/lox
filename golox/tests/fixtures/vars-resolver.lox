
var a = "global";
{
  fun showA() {
    print a;
  }

  showA();
  var a = "block";
  showA();
}

///

fun readVarInInitializer() {
  var x = x;
}

///

fun duplicateLocalDecleration() {
  var y = "first";
  var y = "second";
}
