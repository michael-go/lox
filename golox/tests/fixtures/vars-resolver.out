# exit code: 1
# stdout:
[line 16] Error: : Can't read local variable in its own initializer.
[line 23] Error: : Already a variable with this name in this scope.
global
global
